tests/jax/components/executing/action_selection.py

```diff
@@ -0,0 +1,237 @@
+# python3
+# Copyright 2021 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for FeedforwardExecutorSelectAction class for Jax-based Mava systems"""
+
+from types import SimpleNamespace
+
+import jax
+import pytest
+from acme.types import NestedArray
+
+from mava.components.jax.executing.action_selection import (
+    ExecutorSelectActionConfig,
+    FeedforwardExecutorSelectAction,
+)
+from mava.systems.jax.executor import Executor
+
+
+@pytest.fixture
+def dummy_config() -> ExecutorSelectActionConfig:
+    """Dummy config attribute for FeedforwardExecutorSelectAction class
+
+    Returns:
+        ExecutorSelectActionConfig
+    """
+    config = ExecutorSelectActionConfig()
+    config.parm_0 = 1
+    return config
+
+
+@pytest.fixture
+def dummy_empty_config() -> ExecutorSelectActionConfig:
+    """Dummy empty config attribute for FeedforwardExecutorSelectAction class
+
+    Returns:
+        ExecutorSelectActionConfig
+    """
+    return ExecutorSelectActionConfig()
+
+
+@pytest.fixture
+def mock_empty_executor() -> Executor:
+    """Mock executore component with empty observations"""
+    store = SimpleNamespace(is_evaluator=None, observations={})
+    return Executor(store=store)
+
+
+def get_action(parm_0, parm_1, parm_2):
+    """Function used in the networks.
+
+    Returns:
+        action_info and policy info
+    """
+    return "action_info_after_get_action", "policy_info_after_get_action"
+
+
+class MockExecutor(Executor):
```

Comment:
Keep an eye out for where this can be re-used when testing other executor components. If it can be used in others, then this should go in `mocks.py`

Reply:
Still not sure about this since there is already MockExecutor in mocks.py so probably changing the name and then add it to mocks or make mocks.py just for components